activerecord/lib/active_record/connection_adapters/abstract_adapter.rb

```diff
@@ -93,11 +93,11 @@ def self.build_read_query_regexp(*parts) # :nodoc:
       end
 
       def self.quoted_column_names # :nodoc:
-        @quoted_column_names ||= {}
+        @quoted_column_names ||= Concurrent::Map.new
```

Comment:
Are those caches intended to be per subclass?
Could it be global and initialized eagerly in a constant instead?

That would avoid the race here, which is harmless in the result but can cause extra computations and allocations (e.g. multiple threads creating the Map and creating an entry inside it and then one of the threads overriding the other's map).
(Also if any caller would call this method twice they could see different maps and inconsistency, but it's not so likely a caller would do that)

Reply:
Yeah, we could set it on `inherited`, I'm fine with that, it's also better for MRI shapes.

Reply:
Hum, or we could just remove it from `AbstractAdapter` and just have one constant in each adapter.

Reply:
At least it would be obvious it's a global cache.